jOOQ-codegen/src/main/java/org/jooq/codegen/JavaWriter.java

```diff
@@ -202,7 +202,7 @@ protected String beforeClose(String string) {
 
     @Override
     protected List<String> ref(List<String> clazz, int keepSegments) {
-        List<String> result = new ArrayList<String>();
+        List<String> result = new ArrayList<String>(clazz == null ? 0 : clazz.size());
```

Comment:
Do note that `new ArrayList()` and `new ArrayList(0)` do not behave the same way once the list grows. Irrelevant in this case, but worth mentioning